corehq/apps/users/templates/users/partials/edit_role_modal.html

```diff
@@ -516,7 +516,7 @@ <h4 class="modal-title" data-bind="text: modalTitle"></h4>
             </legend>
 
             {% if has_report_builder_access %}
-              <div class="form-group">
+              <div class="form-group" data-bind="visible: permissions.access_all_locations">
```

Comment:
If a user removes full location access, it looks like this permission will still remain. I think if we need to restrict this only to roles with full access, then we will need to make sure to reset this permission when the other is removed

Reply:
I agree that it should probably behave the way you suggest. But it also looks like the existing permissions that depend on access_all_locations (e.g. Web Users and Groups) are not reset either.

I don't really know Knockout.js, so I'd be reluctant to attempt this in the dialog box itself. It looks like it could be done in the 'post_user_role' function though.

Reply:
I added code to disable permissions for location-restricted users. There was a test failure due to new arguments in a method, so I've synced my branch and added a fix.